src/main/java/seedu/address/model/event/DayPlan.java

```diff
@@ -0,0 +1,120 @@
+import seedu.address.model.event.exceptions.EventClashException;
+import seedu.address.model.event.exceptions.EventNotFoundException;
+
+public class DayPlan {
+    private final ArrayList<Schedule> scheduledTasks;
+
+    public DayPlan() {
+        this.scheduledTasks = new ArrayList<Schedule>();
+    }
+
+    public DayPlan(ArrayList<Schedule> scheduledTasks) {
+        this.scheduledTasks = scheduledTasks;
+    }
+
+    public String addSchedule(Schedule currTask) {
+        if (this.isClash(currTask)) {
+            throw new EventClashException();
+        }
+        this.scheduledTasks.add(currTask);
+        return currTask.toString() + " is added into the timetable";
+    }
+
+    public String deleteSchedule(Schedule currTask) {
+        int currTimeFrom = currTask.getTimeFrom();
+        int currTimeTo = currTask.getTimeTo();
+
+        boolean isRemoved = scheduledTasks
+                .removeIf(task -> (task.getTimeFrom() == currTimeFrom && task.getTimeTo() == currTimeTo));
+        if (!isRemoved) {
+            throw new EventNotFoundException();
+        }
+        return currTask.toString() + " is removed from timetable.";
+    }
+
+    public String viewDayPlan() {
+        if (this.scheduledTasks.size() == 0) {
+            throw new EmptyEventException();
+        }
+        this.scheduledTasks.sort(getComparatorEarlyToLate());
+        String dayPlans = "";
+        for (int i = 0; i < this.scheduledTasks.size(); i++) {
+            dayPlans += this.scheduledTasks.get(i).toString() + "\n";
+        }
+        return dayPlans;
+    }
+
+    public boolean isClash(Schedule currTask) {
+        int currTimeFrom = currTask.getTimeFrom();
+        int currTimeTo = currTask.getTimeTo();
+
+        ArrayList<Schedule> clashList = new ArrayList<Schedule>();
+
+        this.scheduledTasks.stream()
+                .filter(task -> (currTimeFrom > task.getTimeFrom() && currTimeFrom < task.getTimeTo())
+                        || (currTimeTo > task.getTimeFrom() && currTimeTo < task.getTimeTo())
+                        || (task.getTimeFrom() == currTimeFrom) || (task.getTimeTo() == currTimeTo))
+                .forEach(task -> clashList.add(task));
+        return clashList.size() != 0;
+    }
+
+    public String moveSchedule(Schedule from, Schedule to) {
+        DayPlan copyScheduledTasks = new DayPlan(new ArrayList<>(this.scheduledTasks));
+        copyScheduledTasks.deleteSchedule(from);
+
+        if (copyScheduledTasks.isClash(to)) {
+            return "Unable to make changes to the Timetable due to clashes. Please try again.";
+        }
+        this.deleteSchedule(from);
+        this.addSchedule(to);
+        return "Changes have been made successfully.";
+    }
+
+    public void markDone(Schedule currTask) {
+        int index = isExist(currTask);
+        if (index < 0) {
+            return;
+        }
+        this.scheduledTasks.set(index, currTask.markAsDone());
+    }
+
+    public int isExist(Schedule currTask) {
```

Comment:
This method name should be changed to `getScheduleIndex`, `isExist` sounds like it should return a `boolean`.